src/main/java/net/dv8tion/jda/internal/entities/UserById.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2015-2020 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.entities;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.PrivateChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.Contract;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.EnumSet;
+import java.util.List;
+
+public class UserById implements User
+{
+    private final long id;
+
+    public UserById(long id)
+    {
+        this.id = id;
+    }
+
+    @Override
+    public long getIdLong()
+    {
+        return this.id;
+    }
+
+    @Contract("->fail")
+    private void unsupported()
+    {
+        throw new UnsupportedOperationException("This User instance only wraps an ID. Other operations are unsupported");
+    }
+
+    @Nonnull
+    @Override
+    public String getName()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getDiscriminator()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public String getAvatarId()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getDefaultAvatarId()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Nonnull
+    @Override
+    public String getAsTag()
+    {
+        unsupported();
+        return null;
+    }
+
+    @Override
+    public boolean hasPrivateChannel()
+    {
+        unsupported();
+        return false;
```

Comment:
Should we just return false?

Reply:
I'd keep it consistent as the point is that this entity exists exclusively to provide access to the entity. Returning a value, even if it is false, would break that consistency and is technically not true as well.

We don't actually have the info needed to inform true or false, so let's just throw